### ISSUE TITLE
Losing data

### ISSUE THREAD

**Issue by germasch** [ornladios/ADIOS2#1292]
I'll put up the testing code as a PR, so you can check yourself. The tests does use multiblock writing
to write an 10-element array in two pieces.  (It can run in parallel, but the problem happens in serial, too). Using two `Mode::Sync` Puts:

```sh
[kai@macbook build (pr/bug)]$ bin/bpls -dD multi_sync.bp
  double   var   {10}
        step 0:
          block 0: [0:4]
    (0)    0 1 2 3 4
          block 1: [5:9]
    (0)    5 6 7 8 9
```
Looks good. Using two `Mode::Deferred` puts, same result. Good.

Now using a deferred put for one block and a sync put for the other:
```sh
[kai@macbook build (pr/bug)]$ bin/bpls -dD multi_ds.bp
  double   var   {10}
        step 0:
          block 0: [5:9]
    (0)    5 6 7 8 9
```
Oops, ADIOS2 lost half of my data.

**Comment by eisenhauer** [comment]
Thanks Kai...  I suspect there are more vulnerabilities like this and we need to do better regression testing around mixing modes, etc.

**Comment by williamfgc** [comment]
@germasch thanks, let's make those tests proper. I won't probably have time until next week, but I checked with bpls that this fixes this bug.

**Comment by germasch** [comment]
I automated the tests and updated the test PR. This one is resolved. Thanks!
